src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,634 @@
+  // size distribution [cm]
+  // @param [in]  rhoaero     aerosol density [g/cm^3]
+  // @param [in]  temp        temperature [K] real(r8),
+  // @param [in]  press       pressure [dyne/cm^2]
+  // @param [out] scavratenum scavenging rate for aerosol number [1/hour]
+  // @param [out] scavratenum scavenging rate for aerosol volume [1/hour]
+
+  // const Real SHR_CONST_BOLTZ =
+  //     1.38065e-23; //  ! Boltzmann's constant ~ J/K/molecule
+  // const Real SHR_CONST_AVOGAD =
+  //     6.02214e26; //   ! Avogadro's number ~ molecules/kmole
+  // const Real rgas_kmol =
+  //     SHR_CONST_AVOGAD *
+  //     SHR_CONST_BOLTZ; //       ! Universal gas constant ~ J/K/kmole => ! Gas
+  // //                      //       constant (J/K/mol)
+  // const Real rgas = rgas_kmol * 1.e-3 * 1.e7; //
+  // // air molar density [dyne/cm^2/erg*mol = mol/cm^3]
+  // const Real cair = press / (rgas * temp);
+  // air molar density [mol/cm^3]
+  // const Real rhoair = 28.966 * cair;
+
+  const Real pi = haero::Constants::pi;
+  const Real zero = 0;
+  const Real one = 1;
+  const Real two = 2;
+  const Real ten = 10;
+  const Real one_thousand = 1000;
+
+  // local variables
+  Real rrainsv[nrainsvmax] = {zero};    // rain radius for each bin [cm]
+  Real xnumrainsv[nrainsvmax] = {zero}; // rain number for each bin [#/cm3]
+  Real vfallrainsv[nrainsvmax] = {
+      zero}; // rain falling velocity for each bin [cm/s]
+
+  Real raerosv[naerosvmax] = {zero}; // aerosol particle radius in each bin [cm]
+  Real fnumaerosv[naerosvmax] = {
+      zero}; // fraction of total number in the bin [fraction]
+  Real fvolaerosv[naerosvmax] = {
+      zero}; // fraction of total volume in the bin [fraction]
+
+  // this subroutine is calculated for a fix rainrate of 1 mm/hr
+  // precipitation rate, fix as 1 mm/hr in this subroutine [cm/s]
+  const Real precip = one / Real(36000.); //  1 mm/hr in cm/s
+
+  // set the iteration radius for rain droplet
+  // rain droplet bin information [cm]
+  // BAD CONSTANT
+  const Real rlo = .005;
+  // const Real rhi = .250;
+  const Real dr = 0.005;
+  // // Nearest whole number: nint
+  // // number of rain bins
+  // const int nr = 1 + haero::round((rhi - rlo) / dr);
+  // FIXME: values to compute nr are hard-coded.
+  const int nr = 50;
+  // We comment this line because nr is hard-coded.
+  // if (nr > nrainsvmax) {
+  //   printf("subr. calc_1_impact_rate -- nr > nrainsvmax \n ");
+  //   return;
+  // }
+
+  // aerosol modal information
+  // aerosol bin information
+  const Real ag0 = dg0 / two; // mean radius of aerosol
+  // standard deviation (log-normal distribution)
+  const Real sx = haero::log(sigmag);
+  // log(mean radius) (log-normal distribution)
+  const Real xg0 = haero::log(ag0);
+  const Real xg3 = xg0 + Real(3.) * sx * sx; // mean + 3*std^2
+  // BAD CONSTANT
+  // set the iteration radius for aerosol particles
+  const Real dx = haero::max(0.2 * sx, 0.01);
+  const Real xlo = xg3 - haero::max(4. * sx, 2. * dx);
```

Comment:
It confuses me a little for the different ways of using numbers. I have seen numbers defined as `const read two = 2`, use of `Real(3.)` and direct use of` 2.` . Any specific reason to use different ways?

Reply:
Some of the noise comes from typecasting. For example,  `Real(3.)` seems unnecessary, because `3.` is a floating point number already, but C++ interprets constants as double precision, so maybe this expression convert it to single precision where needed. But I don't actually think most of this is necessary unless we see that it causes a problem.

In any case, I don't believe we're following consistent rules for this across the whole team, so I wouldn't worry about it too much. @odiazib or @mjs271 , if I'm oversimplifying, please chime in.

Reply:
Hi @jeff-cohere, you are correct. I like to defined number of const real, if the number appears more than twice. However, I need to be consistent with this approach.

Reply:
@singhbalwinder, here are a few comment about defining numbers.